### PR TITLE
[Foxy backport]: De-deplicate type_support_map.h header

### PR DESCRIPTION
**Pull request by j-rivero**
Port of #81 to Foxy branch.

Testing in foxy CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12304/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7276/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10020/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12202/)

### CODE REVIEW

**Comment by clalancette** [comment]
@j-rivero FYI, I retargeted this to the `foxy` branch and updated the name to have `Foxy backport` in it.

**Comment by j-rivero** [comment]
Merged with foxy branch and fixed the warning in the benchmark test.

File: rosidl_typesupport_cpp/include/rosidl_typesupport_cpp/type_support_map.h [hunk 1]
```diff
@@ -15,38 +15,12 @@
...
-#include <cstddef>
+// Keep this file not to break API. It was no more than
+// a copy of rosidl_typesupport_c/type_support_map.h
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
+#warning rosidl_typesupport_cpp/type_support_map.h header is deprecated \
+  in favor of rosidl_typesupport_c/type_support_map.h.
```
Comment (clalancette) [inline rosidl_typesupport_cpp/include/rosidl_typesupport_cpp/type_support_map.h:22]: Hm, I wonder if we should remove this warning for the Foxy backport.  After all, downstream consumers may already be relying on it.  @jacobperron thoughts?

Comment (jacobperron) [reply rosidl_typesupport_cpp/include/rosidl_typesupport_cpp/type_support_map.h:22]: I agree, we should avoid backporting deprecation warnings.

**Comment by j-rivero** [comment]
> I agree, we should avoid backporting deprecation warnings.

Makes sense, 3603dc2

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12981/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7933/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10708/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12959/)

**Review by jacobperron — APPROVED** [review]

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Pull request merged**